src/test/java/com/google/cloud/spanner/pgadapter/StatementTest.java

```diff
@@ -300,4 +314,71 @@ public void testPortalStatementDescribePropagatesFailure() throws Exception {
 
     intermediateStatement.describe();
   }
+
+  @Test
+  public void testBatchStatements() throws Exception {
+    String sql =
+        "INSERT INTO users (id) VALUES (1); INSERT INTO users (id) VALUES (2);INSERT INTO users (id) VALUES (3);";
+    IntermediateStatement intermediateStatement = new IntermediateStatement(sql, connection);
+
+    Assert.assertTrue(intermediateStatement.isBatchedQuery());
+    List<String> result = intermediateStatement.getStatements();
+    Assert.assertEquals(result.size(), 3);
+    Assert.assertEquals(result.get(0), "INSERT INTO users (id) VALUES (1);");
+    Assert.assertEquals(result.get(1), "INSERT INTO users (id) VALUES (2);");
+    Assert.assertEquals(result.get(2), "INSERT INTO users (id) VALUES (3);");
+  }
+
+  @Test
+  public void testBatchStatementsWithEmptyStatements() throws Exception {
+    String sql = "INSERT INTO users (id) VALUES (1); ;;; INSERT INTO users (id) VALUES (2);";
+    IntermediateStatement intermediateStatement = new IntermediateStatement(sql, connection);
+
+    Assert.assertTrue(intermediateStatement.isBatchedQuery());
+    List<String> result = intermediateStatement.getStatements();
+    Assert.assertEquals(result.size(), 2);
+    Assert.assertEquals(result.get(0), "INSERT INTO users (id) VALUES (1);");
+    Assert.assertEquals(result.get(1), "INSERT INTO users (id) VALUES (2);");
+  }
+
+  @Test
+  public void testBatchStatementsWithQuotes() throws Exception {
+    String sql =
+        "INSERT INTO users (name) VALUES (';;test;;'); INSERT INTO users (name1, name2) VALUES ('''''', ';'';');";
+    IntermediateStatement intermediateStatement = new IntermediateStatement(sql, connection);
+
+    Assert.assertTrue(intermediateStatement.isBatchedQuery());
+    List<String> result = intermediateStatement.getStatements();
+    Assert.assertEquals(result.size(), 2);
+    Assert.assertEquals(result.get(0), "INSERT INTO users (name) VALUES (';;test;;');");
+    Assert.assertEquals(result.get(1), "INSERT INTO users (name1, name2) VALUES ('''''', ';'';');");
+  }
+
+  @Test
+  public void testBatchStatementsWithComments() throws Exception {
```

Comment:
* can we add connection statements (BEGIN; COMMIT) in this test suite too 
* can we also add tests to cover statements that start with "WITH" clause
* I assume we support CREATE | ALTER | DROP DDL statements

Reply:
- WITH expressions are not supported by the backend yet.
- BEGIN and COMMIT are not supported within Cloud Spanner batch statements.
- CREATE | ALTER | DROP are supported.

Reply:
I think its important to test the "not supported" features to make sure we are doing the right behavior, i.e. throwing an error message that is meaningful to the customer

also, WITH is mainly to make sure the PgAdapter Parsing is handling it (not crashing/erroring in weird ways).

Reply:
I added more tests for these other cases.

Reply:
add info in README